### PR TITLE
Bug 1164353 - migrate browser context menu to use smart-modal-dialog

### PR DESCRIPTION
**Pull request by huchengtw-moz**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2c2567fed636cef43d8fd06c49d806d6a238d15c

File: tv_apps/smart-system/js/browser_context_menu.js [hunk 1]
```diff
@@ -61,47 +49,34 @@
...
-
-    // Loop and add element with camel style name to Modal Dialog attribute.
-    this.elementClasses.forEach(function createElementRef(name) {
-      this.elements[toCamelCase(name)] =
-        this.element.querySelector('.' + this.ELEMENT_PREFIX + name);
-    }, this);
+    this.modalDialog = new SmartModalDialog(this.element);
+    // listen the event bubbled from modal dialog to determine the state
+    this.element.addEventListener('modal-dialog-opened', this);
+    this.element.addEventListener('modal-dialog-closed', this);
```
Comment (rexboy7) [inline tv_apps/smart-system/js/browser_context_menu.js]: Should we move them to `_registerElements()`?

Comment (huchengtw-moz) [inline tv_apps/smart-system/js/browser_context_menu.js]: oh~ sure.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=296f4ad1407b9a4e767be98c394a0c3af717fa39

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=05e23fb2901e114e666a9fd1a2afd70af162acc6

**Pull request merged**